### ISSUE TITLE
[MultiSelectTree] Keyboard Navigation issue with Filtering functionality enabled

### ISSUE THREAD

**Issue by georgi-sla** [telerik/kendo-angular#3973]
**Describe the bug**
When the filtering of the MultiSelectTree is enabled and some text is typed in the filter textbox, the keyboard navigation does not behave as expected. The Up and Down arrow keys move the cursor as expected, but the Left and Right arrow keys do not change the position of the cursor as they should. Additionally, when the user tries to select the text in the filter textbox with Shift + Left/Right arrow the text is not being selected.

**To Reproduce**
1. Create a MultiSelectTree component.
2. Enable the filtering functionality. 
3. Use the arrow keys to move the position of the cursor. 

Please, check out the following article and demo: https://www.telerik.com/kendo-angular-ui/components/dropdowns/multiselecttree/filtering/#toc-filtering-with-the-hierarchy-binding-directive

**Expected behavior**
When typing text in the filter textbox and then using the arrow keys to navigate, the cursor should be repositioned in accordance with the [outlined keyboard navigation](https://www.telerik.com/kendo-angular-ui/components/dropdowns/multiselecttree/keyboard-navigation/#toc-tags-navigation). Furthermore, using the combination of Shift + Left/Right arrow key should select the text in the filter textbox.

**Comment by stnikolova** [comment]
The fix is available in version **16.2.0**.
